src/nspawn/nspawn.c

```diff
@@ -848,6 +848,10 @@ static int parse_argv(int argc, char *argv[]) {
                                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                                        "Network interface name not valid: %s", optarg);
 
+                        r = check_interface(optarg);
+                        if (r < 0)
+                                return r;
+
```

Comment:
so we check all those interfaces twice now? Why?

Reply:
I'm not sure what you mean here? The `ARG_NETWORK_INTERFACE` and `ARG_NETWORK_MACVLAN` cases don't seem to fall through. Doesn't that mean we have to repeat the check for all the different types of interfaces? As far as I can see we're checking each network interface once in the switch now and don't check them again after.

Reply:
we check in move_network_interfaces() again, no?

Reply:
`move_network_interfaces` just calls `parse_interface` which doesn't do the check anymore (unless I'm missing something really obvious somewhere). The idea was to move the checking out of `parse_interface` so it could be used from inside a different network namespace without udev complaining.